docs/book/src/tasks/experimental-features/runtime-sdk/implement-topology-mutation-hook.md

```diff
@@ -192,6 +296,97 @@ function openSwaggerUI() {
 }
 </script>
 
+### DiscoverVariables
+
+A DiscoverVariables call returns definitions for one or more variables. This variable definition is the same as that used for inline variables.
+
+Example variable definition:
+
+```yaml
+  - name: etcdImageTag # name of the variable
+    required: true # whether the variable must be defined for Clusters.
+    schema:
+      openAPIV3Schema: 
+        type: string 
+        default: "3.5.3-0" # variable value which is defaulted during reconciliation if not defined elsewhere.
+        example: "3.5.3-0" # example for user guidance.
+        description: "etcdImageTag sets the tag for the etcd image." #description of variable use.
+        # Additional fields from clusterv1.JSONSchemaProps
+```
+
+#### Example Request:
+
+* The request is a simple request which contains the name and namespace of the ClusterClass it is called for.
+TODO: Should this send the full ClusterClass?
+
+```yaml
+apiVersion: hooks.runtime.cluster.x-k8s.io/v1alpha1
+kind: DiscoverVariablesRequest
+clusterClass:
+- name: <clusterClass-name>
+  namespace: <clusterClass-namespace>
+```
+
+#### Example Response:
+
+```yaml
+apiVersion: hooks.runtime.cluster.x-k8s.io/v1alpha1
+kind: DiscoverVariablesResponse
+status: Success # or Failure
```

Comment:
The response should also include `message`. `status` and `message` are required for all response objects.

Reply:
Hm definitely of the API type. But are we actually setting message if the status is Success?

I think originally we planned to only set the message if we have a Failure

Reply:
Hm our lifecycle handlers don't even set the status in the success case. That doesn't seem right :)